app/controllers/applications/process_controller.rb

```diff
@@ -34,6 +35,67 @@ def application_details_save
       end
     end
 
+    def benefits
+      if application.savings_investment_valid?
+        @form = Applikation::Forms::Benefit.new(application)
+        render :benefits
+      else
+        redirect_to application_summary_path(application)
+      end
+    end
+
+    def benefits_save
+      @form = Applikation::Forms::Benefit.new(application)
+      @form.update_attributes(form_params(:benefits))
+
+      if @form.save
+        BenefitCheckRunner.new(application).run
+        redirect_to(action: :benefits_result)
+      else
+        render :benefits
+      end
+    end
+
+    def benefits_result
+      if application.benefits
+        @application = application
+        render :benefits_result
+      else
+        redirect_to(application_build_path(application_id: application.id, id: :income))
+      end
+    end
+
+    def income
+      if !application.benefits?
```

Comment:
Could we maybe add `.income?` to Application model, so that this reads 'better'?

It would essentially be the same as above:

```ruby
def income?
  !benefits?
end
```